### PR TITLE
Feature/ab#14 detail view care users

### PR DESCRIPTION
**Pull request by BerendWouters**
Introduces detail view for care users

### CODE REVIEW

File: Singer.API/ClientApp/src/app/modules/core/models/careuser.model.ts [hunk 1]
```diff
@@ -9,12 +11,60 @@ export interface PaginationDTO{
...
+
+export interface UpdateCareUserDTO {
+   firstName: string;
+   lastName: string;
    email: string;
    userName: string;
    birthday: Date;
    caseNumber: string;
-   ageGroup: string; //Maybe replace by own class?
+   ageGroup: AgeGroup; //Maybe replace by own class?
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/core/models/careuser.model.ts]: Is this change not the implementation of our own class? If so, please remove comment

File: Singer.API/ClientApp/src/app/modules/core/services/care-users-api/careuser.proxy.ts [hunk 2]
```diff
@@ -21,4 +22,12 @@ export class CareUserProxy{
       return this.apiService.get('api/careuser', searchParams).pipe(map((res) => res));
    }
 
+   updateCareUser(id: string, updateCareUserDTO: UpdateCareUserDTO) {
+   return this.apiService.put(`api/careuser/${id}`, updateCareUserDTO).pipe(map((res) => res));
```
Comment (vyruz1986) [inline Singer.API/ClientApp/src/app/modules/core/services/care-users-api/careuser.proxy.ts]: Indentation wrong?

File: Singer.API/DTOs/CreateUserDTO.cs [hunk 3]
```diff
@@ -5,12 +5,19 @@ namespace Singer.DTOs
 {
    public class CreateUserDTO
    {
+
+      [Required]
+      [StringLength(maximumLength: 255,
+         ErrorMessage = "De {0} moet een lengte hebben van minstens {2} en maximum {1} karakters.",
+         MinimumLength = 3)]
+      [DisplayName("FirstName")]
```
Comment (vyruz1986) [inline Singer.API/DTOs/CreateUserDTO.cs]: should be 'Voornaam'?

File: Singer.API/Services/MockUserService.cs [hunk 4]
```diff
@@ -24,7 +25,7 @@ public class MockUserService : IUserService
          {
             Id = Guid.NewGuid(),
             User = new User() {
-               Name = "Joske Vermeulen"
+               LastName = "Joske Vermeulen"
```
Comment (vyruz1986) [inline Singer.API/Services/MockUserService.cs]: This (as well as the next mock users) don't have a first name...

File: Tests/ServiceTests/StringFilterTests.cs [hunk 5]
```diff
@@ -21,7 +21,7 @@ public void GetFilterExpressionExactPropertyValue()
             new CareUser
             {
                Id = Guid.NewGuid(),
-               User = new User(){Name = "Joske Vermeulen"},
+               User = new User(){LastName = "Joske Vermeulen"},
```
Comment (vyruz1986) [inline Tests/ServiceTests/StringFilterTests.cs:24]: This (as well as the next users) user has no first name

**Review by vyruz1986 — CHANGES_REQUESTED** [review]
Remarks:
* All material imports should be done in `material.module.ts`, this class exports `MaterialModule`, which can then be imported in other components. This keeps all material component imports centrally configured.
* `care-user-details.component.ts` has separate functions for every field, most of them do exactly the same...

File: Singer.API/ClientApp/src/app/modules/core/models/careuser.model.ts (hunk 1 above)
Comment (BerendWouters) [reply Singer.API/ClientApp/src/app/modules/core/models/careuser.model.ts]: Yes, yet it is an enum.

**Comment by Joren-Thijs** [comment]
Imports in material.module.ts are done as requested.
Different getFieldErrorMessages are replaced by more abstract method as requested.
Old ageGroup comments in careuser.model.ts are removed as requested.

**Review by vyruz1986 — APPROVED** [review]

**Pull request merged**